File: static/CWE-94/python/vulnerable.py
# CWE-94: Code Injection
# User-supplied input is evaluated as Python code.
# Vulnerable sink: eval()

from flask import Flask, request

app = Flask(__name__)


@app.route("/calc")
def calc():
    expression = request.args.get("expr")
    result = eval(expression)
    return str(result)

File: static/CWE-94/javascript/vulnerable.js
// CWE-94: Improper Control of Generation of Code ('Code Injection')
// Untrusted request input is passed directly to eval() and executed as code.
// Vulnerable sink: eval(req.query.expr)

const express = require('express');

const app = express();

app.get('/calc', (req, res) => {
  const expr = req.query.expr;
  const result = eval(expr);
  res.send(String(result));
});

app.listen(3000);
